plugins/woocommerce/src/StoreApi/Utilities/AgenticCheckoutUtils.php

```diff
@@ -332,26 +332,83 @@ public static function add_protocol_headers( \WP_REST_Response $response, \WP_RE
 	}
 
 	/**
-	 * Check if the Agentic Checkout feature is enabled.
+	 * Check if the Agentic Checkout feature is enabled and request is authorized.
 	 *
-	 * V1 implementation: Returns true if feature is enabled (no auth check).
-	 * Future: Implement Bearer token authentication.
+	 * Validates bearer token against registered agents in the agent registry.
 	 *
+	 * @param \WP_REST_Request $request Request object.
 	 * @return bool|\WP_Error True if authorized, WP_Error otherwise.
 	 */
-	public static function is_authorized() {
-		// Check if feature is enabled.
-		$features_controller = wc_get_container()->get( FeaturesController::class );
-		if ( ! $features_controller->feature_is_enabled( 'agentic_checkout' ) ) {
+	public static function is_authorized( $request = null ) {
+		if ( null === $request ) {
 			return new \WP_Error(
-				'woocommerce_rest_agentic_checkout_disabled',
-				__( 'Agentic Checkout API is not enabled.', 'woocommerce' ),
-				array( 'status' => 403 )
+				'invalid_request',
+				__( 'Invalid request object.', 'woocommerce' ),
+				array(
+					'status' => 400,
+					'type'   => 'invalid_request',
+					'code'   => 'invalid_request',
+				)
+			);
+		}
+
+		$auth_header = $request->get_header( 'Authorization' );
+		if ( empty( $auth_header ) || 0 !== stripos( $auth_header, 'Bearer ' ) ) {
+			return new \WP_Error(
+				'invalid_request',
+				__( 'Invalid authorization.', 'woocommerce' ),
+				array(
+					'status' => 400,
+					'type'   => 'invalid_request',
+					'code'   => 'invalid_authorization_format',
+				)
+			);
```

Comment:
Maybe an improvement in the next PR. This is not a blocker for this PR. 

I think all WP_Error can use `Error` and then have a method `to_wp_error`, similar to the current `to_rest_response`. Here is the pattern for using it. The rationale for `Error` class is to help us ensure the response shape in accordance with the specs.

https://github.com/woocommerce/woocommerce/blob/update%2Fagentic-checkout-auth/plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessionsComplete.php#L300-L301